trojsten/special/plugin_prask_5_1_1/views.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+from django.contrib.auth.decorators import login_required
+from django.core.urlresolvers import reverse
+from django.http import HttpResponse, HttpResponseBadRequest
+from django.shortcuts import render
+
+from .core import LEVELS
+from .models import UserLevel
+from .update_points import update_points
+
+
+@login_required()
+def main(request, level=1):
+    level = max(min(int(level), 10), 1)
+    user = request.user
+    userlevel, _ = UserLevel.objects.get_or_create(level_id=level, user=user)
+
+    target = LEVELS[level].TARGET
+
+    try_set = []
+    for x in userlevel.try_set.order_by('id'):
+        try_set.append((x.input, x.output, x.output == target))
+
+    levels = [[i, False] for i in range(1, 11)]
+    for x in UserLevel.objects.filter(user=user):
+        levels[x.level_id - 1][1] = x.solved
+
+    return render(request, 'plugin_prask_5_1_1/level.html', {
+        "level": level,
+        "levels": levels,
+        "solved": userlevel.solved,
+        "target": target,
+        "try_set": try_set,
+        "try_count": userlevel.try_count,
+        "try_count_ending":
+        {1: '', 2: 'y', 3: 'y', 4: 'y'}.get(userlevel.try_count, 'ov'),
+        "maximum": LEVELS[level].MAXIMUM if hasattr(LEVELS[level], 'MAXIMUM') else 9999999999,
+    })
+
+
+@login_required()
+def run(request, level=1):
+    level = max(min(int(level), 10), 1)
+    user = request.user
+    userlevel, _ = UserLevel.objects.get_or_create(level_id=level, user=user)
+
+    try:
+        data = json.loads(request.read().decode("utf-8"))
+        _input = int(data["input"])
+    except (KeyError, ValueError):
+        return HttpResponseBadRequest()
+
+    if _input < 0 or _input >= 10**10:
```

Comment:
Magické konštanty

Reply:
They are not so magical, and as they are used only in that file I am fine keeping them there.

Reply:
But why 10**10? What is this number?

Reply:
When the input has more than 10 digits

Reply:
I've put it into MAX_INPUT variable.